lib/response-stream.js

```diff
@@ -0,0 +1,37 @@
+import { Readable } from 'node:stream';
+
+export class ResponseStream extends Readable {
+    constructor(options) {
+        super(options);
+        this.buffer = [];
+        this.isReading = false;
+    }
+
+    // Method to add data to the internal buffer
+    // This method simply adds data; the stream will request it when needed
+    send(data) {
+        this.buffer.push(data);
+        if (this.isReading) {
+            this.isReading = false;
+            this._read();
+        }
+    }
+
+    // _read method, automatically called by the stream when it wants more data
+    _read() {
+        if (this.buffer.length > 0) {
+            const chunk = this.buffer.shift(); // Get the next chunk from the buffer
+            this.push(chunk); // Push the chunk into the stream
+        } else {
+            this.isReading = true;
+        }
+    }
+
+    done() {
+        this.emit('done');
+    }
+
+    end() {
+        setTimeout(() => this.push(null), 0);
+    }
+}
```

Comment:
This Readable stream class probably belongs in the Podium utils package. It provides a .send() method to push chunks of html onto the response, a .done() method to be used by the developer to signal when they are finished pushing content, as well as a .end() method to indicate that the app has no more html to push and closes the stream.

Essentially, html template uses .send() to push the document head, then the application developer uses .send() to push the content and calls .done() when they are finished at which point HTML template takes over again and uses .send() to push the final bits of the template and ends the stream with .end()